src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -159,7 +159,7 @@ int sd_dhcp6_client_set_callback(
 int sd_dhcp6_client_set_ifindex(sd_dhcp6_client *client, int ifindex) {
 
         assert_return(client, -EINVAL);
-        assert_return(ifindex >= -1, -EINVAL);
+        assert_return(ifindex > 0, -EINVAL);
```

Comment:
I wonder if the idea wasn't to allow passing -1 to unset the index. But if there are no uses of that, then indeed rejecting -1 sounds fine.

Reply:
At least networkd does not reset ifindex. So, I reject that in this commit.